src/vm/typeparse.cpp

```diff
@@ -1500,27 +1500,30 @@ DomainAssembly * LoadDomainAssembly(
     StackScratchBuffer buffer;
     LPCUTF8 szAssemblySpec = psszAssemblySpec->GetUTF8(buffer);
     IfFailThrow(spec.Init(szAssemblySpec));
-    
+
     if (spec.IsContentType_WindowsRuntime())
     {
         _ASSERTE(pssOuterTypeName != NULL);
         spec.SetWindowsRuntimeType(*pssOuterTypeName);
     }
-    
-    if (pPrivHostBinder)
-    {
-        spec.SetHostBinder(pPrivHostBinder);
-    }
-    else if (pRequestingAssembly && (!pRequestingAssembly->IsCollectible()))
```

Comment:
Just curious - you removed the condition about collectible assemblies... is that intentional? If so, is it safe - so we know why it was there before?

Reply:
@janvorli Removed the similar structure for `Assembly::Load` in #22494.  This should have been removed at that time, but it was missed.  It would cause the same issue in `Type.GetType()` that #22494 fixed.  I believe it is safe, because it is already done in `Assembly::Load`

Reply:
OK - makes sense.